### PR TITLE
[FEAT] Initial node state - part 2 Adapter changes

### PR DESCRIPTION
**Pull request by sindhu-106**
Signed-off-by: sindhuja <sindhuja.gajula@intel.com>

### CODE REVIEW

File: provisioners/src/main/java/com/intel/dai/provisioners/NetworkListenerProviderForeignBus.java [hunk 1]
```diff
@@ -340,6 +337,8 @@ private PropertyDocument sendForeignApiRequest(String url, String argument) {
 
             if(client_ == null) {
```
Comment (paul-amonson) [inline provisioners/src/main/java/com/intel/dai/provisioners/NetworkListenerProviderForeignBus.java:338]: This code is duplicated, should it be a separate method?

File: provisioners/src/main/java/com/intel/dai/provisioners/NetworkListenerProviderForeignBus.java [hunk 2]
```diff
@@ -367,6 +366,8 @@ private PropertyDocument sendForeignApiRequest(String url) {
...
                 client_ = createClient();
+                if (client_ == null)
+                    throw new RESTClientException("Failed to get the REST client implementation");
                 createTokenProvider();
```
Comment (paul-amonson) [inline provisioners/src/main/java/com/intel/dai/provisioners/NetworkListenerProviderForeignBus.java:371]: Duplicated block.

File: docker-deploy/hw2-config/ProviderProvisionerNetworkForeignBus.json [hunk 3]
```diff
@@ -45,6 +45,16 @@
       "bootImageInfoUrl" : "http://api-gw-service-nmn.local:8080/apis/ims/images",
       "bootImageForImageIdInfoUrl" : "http://api-gw-service-nmn.local:8080/apis/ims/images/",
       "publish": true
+    },
+    "com.intel.dai.provisioners.AdapterProvisionerApi": {
```
Comment (yhuan47) [inline docker-deploy/hw2-config/ProviderProvisionerNetworkForeignBus.json:49]: The comment is actually for the /apis/ims/images URL. I don't think port 8080 is required in the URL. Do we test the URL with port 8080?

Comment (paul-amonson) [reply docker-deploy/hw2-config/ProviderProvisionerNetworkForeignBus.json:49]: Yes. Hw1-3 do not use port 8080

Comment (sindhu-106) [reply docker-deploy/hw2-config/ProviderProvisionerNetworkForeignBus.json:49]: Yes Leo, you are right. 8080 should be removed

File: provisioners/src/main/java/com/intel/dai/provisioners/NetworkListenerProviderForeignBus.java (hunk 1 above)
Comment (sindhu-106) [reply provisioners/src/main/java/com/intel/dai/provisioners/NetworkListenerProviderForeignBus.java]: Yes Paul, moved to separate method

**Review by yhuan47 — APPROVED** [review]

File: provisioners/src/main/java/com/intel/dai/provisioners/NetworkListenerProviderForeignBus.java [hunk 4]
```diff
@@ -388,6 +359,27 @@ private PropertyDocument sendForeignApiRequest(String url) {
...
+
+    private void checkAndCreateClient() throws RESTClientException {
+        if(client_ == null) {
+            client_ = createClient();
+            if (client_ == null)
+                throw new RESTClientException("Failed to get the REST client implementation");
+            createTokenProvider();
+        }
+    }
+
```
Comment (paul-amonson) [inline provisioners/src/main/java/com/intel/dai/provisioners/NetworkListenerProviderForeignBus.java:382]: Good, despite me losing internet you got it!  :)

**Review by paul-amonson — APPROVED** [review]

**Pull request merged**